src/backend/access/transam/xact.c

```diff
@@ -3026,7 +3026,7 @@ CommitTransaction(void)
 	 * worry about aborts as we release session level locks automatically during
 	 * an abort as opposed to a commit.
 	 */
-	if(Gp_role == GP_ROLE_DISPATCH)
+	if(Gp_role == GP_ROLE_DISPATCH || Gp_role == GP_ROLE_SINGLENODE)
```

Comment:
GP_ROLE_SINGLENODE has properties GP_ROLE_DISPATCH and GP_ROLE_UTILITY. Can they be transformed into bit fields?

Reply:
Do you mean `GP_ROLE_SINGLENODE = GP_ROLE_DISPATCH & GP_ROLE_UTILITY` ?
But they are original enum? Do you mean we can change it to int?

Reply:
After consider a while, I think it's unneccessary. The reason is that GP_ROLE_SINGLENODE is not always have all properties of both GP_ROLE_DISPATCH and GP_ROLE_UTILITY.
So that even we make it to int, we still need to mix the style of `Gp_role & GP_ROLE_SINGLENODE` and `Gp_role == GP_ROLE_SINGLENODE`. So the benefit won't be large enough to make changes to the original way of GP_ROLE enum expression.